src/config.ts

```diff
@@ -21,18 +21,49 @@ export const Config = z.object({
 	token: z.string().optional()
 });
 
+export const Cache = z.record(
+	z.object({
+		plan: z.string()
+	})
+);
+
+export type Cache = z.infer<typeof Cache>;
+
 export type Config = z.infer<typeof Config>;
 
+const loadCache = async (path = cacheFilePath): Promise<Cache> => {
+	const caches = parse(await loadFile(path));
+	return Cache.parse({
+		...caches
+	});
+};
+
 const defaultConfig: Config = {
 	baseURL: 'https://dashboard.metacall.io',
 	apiURL: 'https://api.metacall.io',
 	devURL: 'http://localhost:9000',
 	renewTime: 1000 * 60 * 60 * 24 * 15
 };
 
+export const updateCache = async (
+	key: string,
+	val: string,
+	path = cacheFilePath
+) => {
+	const cache = await loadCache(path);
+	cache[key] = { plan: val };
```

Comment:
@viferga wdyt?
can you change the caching preferences to general key-value pairs, this is again the same stuff, CLI should not be limited to caching plans, yes, I know there are not many things right now to cache but in the future, we may, so changes I could suggest is.

Pass key-value pair for caching.
for the above example ```key=plan``` and ```val=planValue```

Reply:
I'm doing what you're describing. Here cache is a record of a project directory as key and an object as value.

That object (value) can contain many things.

But so far, it only contains plan as key.

Reply:
aren't you hard coding only plan?

Reply:
> @viferga wdyt? Great, Can you just change the caching preferences to general key-value pairs, this is again the same stuff, CLI should not be limited to caching plans, yes, I know there are not many things right now to cache but in the future, we may, so changes I could suggest is.
> 
> Pass key-value pair for caching. for the above example `key=plan` and `val=planValue`

I guess we can do that, but then we might loose type safety.

We can make keys optional but still, we need to know what keys are we taking about

Auth token, name, password, e-mail address are stored in base config, so no point of including them in cache.

Reply:
Yeah exactly, I guess creation an interface might solve the problem, like it will work as schema for cache.ini, wdyt?

Reply:
Well, it doesn't look like schema is the problem here.

Zod is fine here as it provides optional keys. But what keys? We still need them in interfaces as well.

The problem is what keys do we need other than plan.

Can we enumerate what else to cache?🤔

Reply:
That's I am saying from starting that for now plan is the only thing to be cached, but previously we were also caching env vara, but as @viferga said we will cache it on db then that's also not an option here but we might encounter some extra preferences to be cached in the future, so we don't have to tweak the code much that's what I meant by scalable, but anyways let's wait what @viferga says.